sormas-app/app/src/main/java/de/symeda/sormas/app/component/dialog/LocationDialog.java

```diff
@@ -106,8 +108,21 @@ protected void initializeContentView(ViewDataBinding rootBinding, ViewDataBindin
 
 		InfrastructureHelper.initializeHealthFacilityDetailsFieldVisibility(contentBinding.locationFacility, contentBinding.locationFacilityDetails);
 
+		if (data.getCountry() == null) {
+			for (Item countryItem : initialCountries) {
+				Country country = (Country) countryItem.getValue();
+				if (country != null && ConfigProvider.isConfiguredServer(country.getIsoCode())) {
```

Comment:
I don't think this code can work, because the country ISO codes are three-digit codes while our locales are either two-digit codes in case there's no country specified (e.g. "en") or have a completely different format (e.g. "de-CH"). Also, it's theoretically possible that a server in a specific country might use a different locale, for whatever reason.

I think the best solution here is to use the `country.name` property that already exists, and compare its value with the name of the country. For this, you need to make that property accessible in InfoResource and retrieve it in the Android app, similar to how the locale is currently retrieved.

Reply:
Country name is marked as deprecated in sormas.properties. There are many things to consider here. Although we reserved 3 characters for ISO code in our database, I had no problem to create a country with only 2 letters for ISO code. That wouldn't be a problem, since there are two standards for ISO codes, one which uses two letters and one which uses three. If I were to choose a solution, I would vote to enforce ISO code to be two letters long, and keep the current implementation. The name of the country doesn't seem that robust to me (what if we localise them in the app and have them in English in sormas.properties?).

Reply:
Yeah, that's true, because it wasn't really needed anymore until now. It's not a problem to make it "undeprecated" again though if we have another use for it. The problem with the ISO code is that we provide a default list of countries that come with SORMAS, and those use the three-digit ISO codes, so that's already the standard that we can't just change all of a sudden because it would break all running servers that have already imported our default country list (or imported their own country list and used three-digit codes).

You're right that using the country name is language-specific, but that would then be something the server administrators would need to consider. Since they're also responsible for adding the countries to their application, they could ensure that the name of the country in SORMAS and the name of the country in the properties file are identical. Localisation is not a problem in that regard because it's taken care of in a separate localisation file, so server admins can choose whether to a) import the default countries in English and then rely on the localisation file to translate these countries into the languages that their users are using or b) import/create their own countries either in English or already in their language, which would then however make the localisation file unusable. In any of these two cases, relying on the country name in this issue would not be a problem because server admins could ensure that it matches.